### PR TITLE
[ua-parser-js] Allow call on constructor fuction to retrieve result

### LINKED ISSUE #52899: ua-parser-js: UAParser no-new usage not covered
UAParser can be used as a function that returns `ReturnType<UAParser[getResult]>`. That usage is not covered by `@types/ua-parser-js`.

```JS
var result = UAParser(uastring1);
```

On the side note, the current declaration is also missing the `new UAParser(extensions)` / `UAParser(extensions)` signature that is present in the original JS library.

see:
- https://github.com/faisalman/ua-parser-js/issues/423
- https://github.com/faisalman/ua-parser-js#constructor
---

- [x] I tried using the `@types/ua-parser-js` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I understand that `@types` package versions [do not follow SemVer](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @superduper @legendecas @MeLlamoPablo @peterblazejewicz

### PR DESCRIPTION
**Pull request by BendingBender**
Fixes #52899 

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/faisalman/ua-parser-js/blob/master/src/ua-parser.js#L753-L755
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/ua-parser-js/ua-parser-js-tests.ts [hunk 1]
```diff
@@ -45,9 +50,21 @@ result.cpu.architecture; // $ExpectType string | undefined
 parser.getCPU().architecture; // $ExpectType string | undefined
 
 // Extensions
-const uaString = 'ownbrowser/1.3';
+const uaString = "ownbrowser/1.3";
 const ownBrowser = [[/(ownbrowser)\/([\w\.]+)/i], [UAParser.BROWSER.NAME, UAParser.BROWSER.VERSION]];
-parser = new UAParser(uaString, { browser: ownBrowser }); // $ExpectType UAParser
+new UAParser({ browser: ownBrowser }); // $ExpectType UAParserInstance
```
Comment (ippeiukai) [inline types/ua-parser-js/ua-parser-js-tests.ts:55]: @BendingBender Thank you for working on this!

I’m not sure if I’m reading this correctly, but wouldn’t this break backward compatibility with existing code like `const parser: UAParser = new UAParser()`?
Could for example `declare type UAParser = UAParserInstance;` coexist with `declare const UAParser` in .d.ts?

Comment (BendingBender) [reply types/ua-parser-js/ua-parser-js-tests.ts:55]: Yeah, you're right, fixed this.

**Review by ippeiukai — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]
LGMT!
@BendingBender thx!

**Comment by BendingBender** [comment]
Ready to merge

**Pull request merged**